pymc3/variational/approximations.py

```diff
@@ -570,6 +570,26 @@ def __init__(self, trace=None, size=None, **kwargs):
             raise opvi.LocalGroupError('Empirical approximation does not support local variables')
         super(Empirical, self).__init__(trace=trace, size=size, **kwargs)
 
+    def evaluate_over_trace(self, node):
```

Comment:
I do not like the name very much, another candidate was `integrate` but does not show what does the the function do

Reply:
No, I leave that for user. He may need variance or other stuff

Reply:
Variance is an expectation :).

Reply:
How about:
```suggestion
    def monte_carlo_expectation(self, node):
```

Reply:
I do not like it. Even if variance is computed as an expectation that is not practical to manually pass the correct expression to integrate over the trace.

Reply:
currently the method just evaluates node over the trace and returns every evaluation

Reply:
Oh ok, then `evaluate_over_trace` is good (or maybe `evaluate_over_points`?)

Reply:
I think trace is more intuitive